### PR TITLE
feat: stored by as dimension [DHIS2-4902]

### PR DESCRIPTION
**Pull request by gnespolino**
This PR adds support for:

- `...&dimension=sb:system;tracker&...` will add `Stored by` in response grid and filter by `storedby in ('system','tracker')`
- `...&dimension=sb&...` will add `Stored by` in response grid without filtering
 
I'll update the documentation as well.

### CODE REVIEW

**Comment by larshelge** [comment]
I think the question here is, do we really want this as an aggregate event analytics dimension? I think that will be strange. It does fit great as another column in event listing mode though. If just event listing, I don't think we need the new interface.

**Comment by gnespolino** [comment]
> I think the question here is, do we really want this as an aggregate event analytics dimension? I think that will be strange. It does fit great as another column in event listing mode though. If just event listing, I don't think we need the new `StoredByDimensionalItemObject.java ` interface.

as far as I understood they want to count grouping by storedBy in a given period.
I feel something is missing as I don't still understand analytics fully.
More than this when they say "in a given period" they probably refer to "createdAt" rather than analytics period, which is instead based on event timestamp. There's another ticket about adding "lastUpdated" (but I think createdAt would be more reasonable) but I don't know how to implement it.

**Review by stian-sandvold — APPROVED** [review]

**Comment by gnespolino** [comment]
> I think the question here is, do we really want this as an aggregate event analytics dimension? I think that will be strange. It does fit great as another column in event listing mode though. If just event listing, I don't think we need the new `StoredByDimensionalItemObject.java ` interface.

Hello @larshelge , I changed this PR to have stored by just as an additional column in the query, please review it when possible.

**Review by maikelarabori — APPROVED** [review]
Looks good :+1:

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcEventAnalyticsTableManager.java [hunk 1]
```diff
@@ -115,6 +117,7 @@ public JdbcEventAnalyticsTableManager( IdentifiableObjectManager idObjectManager
         new AnalyticsTableColumn( quote( "completeddate" ), TIMESTAMP, "psi.completeddate" ),
         new AnalyticsTableColumn( quote( "created" ), TIMESTAMP, "psi.created" ),
         new AnalyticsTableColumn( quote( "lastupdated" ), TIMESTAMP, "psi.lastupdated" ),
+        new AnalyticsTableColumn( quote( STORED_BY_COL_NAME ), CHARACTER_50, "psi.storedby" ),
```
Comment (d-bernat) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcEventAnalyticsTableManager.java:120]: there is a bit different definition for column psi.storedby (programstageinstance table) in 2.37 database: 
storedby character varying(255) COLLATE pg_catalog."default".  Make it sence to use CHARACTER_255?

**Review by d-bernat — CHANGES_REQUESTED** [review]

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcEventAnalyticsTableManager.java (hunk 1 above)
Comment (gnespolino) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcEventAnalyticsTableManager.java:120]: Good catch. I'll change this to `VARCHAR(255)`

**Review by d-bernat — APPROVED** [review]

**Comment by larshelge** [comment]
Great, thanks @gnespolino ! Yes I think this is a natural and nice solution. Approved.

**Review by larshelge — APPROVED** [review]

**Pull request merged**